pyblish_lite/control.py

```diff
@@ -58,40 +66,107 @@ def __init__(self, parent=None):
             "ordersWithError": set()
         }
 
-    def reset(self):
-        """Discover plug-ins and run collection"""
+    def prepare_for_reset(self):
+        self.validated = False
+        self.extracted = False
+        self.publishing = False
+
         self.context = pyblish.api.Context()
-        self.plugins = pyblish.api.discover()
+        self.context.data["optional"] = False
+        self.context.data["publish"] = True
 
-        self.was_discovered.emit()
+        self.context.data["label"] = 'Context'
+        self.context.data["name"] = 'Context'
+        self.context.data["_type"] = 'context'
 
-        self.pair_generator = None
-        self.current_pair = (None, None)
+        self.context.data["icon"] = 'book'
+
+        self.context.data["_has_succeeded"] = False
+        self.context.data["_has_failed"] = False
+        self.context.data["_is_idle"] = True
+
+        self.context.data["__families__"] = ('__context__',)
+
+    def reset(self):
+        """
```

Comment:
Please use the first line for a broad one-line description, and leave a space after the last line, and after the docstring.

```py
def function():
        """Discover plug-ins and run collection

        - prepare_for_reset should be run before
            - was split because of `Context` in instance list
            - reset logic should be used only in window.py in `reset` method

        """

        # This is just backup
        ...
```